src/threads.c

```diff
@@ -25,25 +25,7 @@ const char *netdata_thread_tag(void) {
 // compatibility library functions
 
 pid_t gettid(void) {
-#ifdef __FreeBSD__
-
-    return (pid_t)pthread_getthreadid_np();
-
-#elif defined(__APPLE__)
-
-    #if (defined __MAC_OS_X_VERSION_MIN_REQUIRED && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
-        uint64_t curthreadid;
-        pthread_threadid_np(NULL, &curthreadid);
-        return (pid_t)curthreadid;
-    #else /* __MAC_OS_X_VERSION_MIN_REQUIRED */
-        return (pid_t)pthread_self;
-    #endif /* __MAC_OS_X_VERSION_MIN_REQUIRED */
-
-#else /* __APPLE__*/
-
-    return (pid_t)syscall(SYS_gettid);
-
-#endif /* __FreeBSD__, __APPLE__*/
+    return (pid_t)pthread_self;
```

Comment:
I spelunked through old OS X header files to confirm that the `pthread_self()` function exists prior to 10.6. Assuming it works, `pthread_self()` is part of the POSIX standard, and this nest of `#ifdef`s can be entirely removed!